### PR TITLE
Search api docs

### PR DESCRIPTION
**Pull request by mattss**
@mgrbyte and @adamcheasley here is our stab at the API interface/docs. Let us know if it makes sense and if we've missed anything obvious.

### CODE REVIEW

File: src/ploneintranet/search/interfaces.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    highlighted_summary = Attribute(
+        _(u'A highlighted summary provided by the backend'))
+
+
+class ISearchResponse(Interface):
+    """
+    Interface defining a common response object parsed from search engines
+    backends
+    """
+    results = Attribute(_(u'The SearchResults returned from a query'))
```
Comment (adamcheasley) [inline src/ploneintranet/search/interfaces.py:87]: It's not entirely clear from this interface, but I'm assuming that the results attribute will contain an iterable of ISearchResult objects?

File: src/ploneintranet/search/README.rst [hunk 2]
```diff
@@ -0,0 +1,24 @@
+Plone Intranet Search
+=====================
+
+This package aims to provide a common API
+for indexing and querying from Plone
+to a 3rd party search engines.
```
Comment (adamcheasley) [inline src/ploneintranet/search/README.rst]: Grammatical error here. I believe this should read either:
"to a 3rd party search engine"
or
"to 3rd party search engines"

File: src/ploneintranet/search/interfaces.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    """
+    fields = Attribute(_(u'A dictionary of indexed fields and their values'))
+    highlighted_summary = Attribute(
+        _(u'A highlighted summary provided by the backend'))
+
+
+class ISearchResponse(Interface):
+    """
+    Interface defining a common response object parsed from search engines
+    backends
```
Comment (adamcheasley) [inline src/ploneintranet/search/interfaces.py]: There's either an apostrophe missing here, or one of 'engines' or 'backends' should not be pluralised.

File: src/ploneintranet/search/interfaces.py (hunk 1 above)
Comment (mattss) [inline src/ploneintranet/search/interfaces.py:87]: Yes indeed. Ben C says he knows how to make this clearer. Watch this space.

File: src/ploneintranet/search/interfaces.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
+""" Interfaces for search API
+"""
+from zope.interface import Interface, Attribute
+
+from . import _
+
+
+class IGlobalSearch(Interface):
```
Comment (mgrbyte) [inline src/ploneintranet/search/interfaces.py]: Unsure what  'Global' conveys in this name

File: src/ploneintranet/search/interfaces.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        :param keywords: The keywords to search for
+        :type keywords: str
+        :param facets: The facets to filter results by
+        :type facets: dict
+        :param start: The offset position in results to start from
+        :type start: int
+        :param count: The maximum number of results to return
+        :type count: int
+        :returns: The results as a `SearchResponse` object
+        :rtype: `SearchResponse`
```
Comment (mgrbyte) [inline src/ploneintranet/search/interfaces.py]: Is this meant to be  `returns The search results.` and `rtype: ISearchResponse`?

File: src/ploneintranet/search/interfaces.py (hunk 4 above)
Comment (mattss) [inline src/ploneintranet/search/interfaces.py]: 'Search' was too broad a term to describe what we are doing here. This is designed to represent the user stories for a global / site-wide search, so we tried to get that into the name.

File: src/ploneintranet/search/interfaces.py (hunk 5 above)
Comment (mattss) [inline src/ploneintranet/search/interfaces.py]: Yes indeed. Fix in the works.

File: src/ploneintranet/search/interfaces.py (hunk 4 above)
Comment (mgrbyte) [inline src/ploneintranet/search/interfaces.py]: Fair enough. Would prefer `ISiteSearch` in this case :)

Comment (mattss) [inline src/ploneintranet/search/interfaces.py]: Yeah that's what I wanted but @wengole said no. @adamcheasley you have the deciding vote.

Comment (adamcheasley) [inline src/ploneintranet/search/interfaces.py]: I would go for ISiteSearch

**Pull request merged**